File: latebindingapi/LateBindingApi.CodeGenerator.CSharp/TestClient/Form1.cs
using System;
using System.Reflection;
using System.Collections.Generic;
using System.ComponentModel;
using System.Text;
using System.Windows.Forms;
using stdole;

using LateBindingApi.Core;
using xyz;
namespace ClientApplication
{
    public class Form1 : System.Windows.Forms.Form
    { 
        /// <summary>
        /// Required designer variable.
        /// </summary>
        private System.ComponentModel.IContainer components = null;

        public Form1()
        {
            InitializeComponent();

            /*Initialize LateBindingApi*/
            LateBindingApi.Core.Factory.Initialize();
            // LateBindingApi.Core.Settings.EnableEvents = true;
            
            /*>> your testcode here <<*/
            Word.Application wordApp = new Word.Application();
            wordApp.Visible = true;
            foreach (Office.COMAddIn item in wordApp.COMAddIns)
                Console.WriteLine(item.Description);

            wordApp.Quit();
            wordApp.Dispose();

            Excel.Application exApp = new LateBindingApi.ExcelApi.Application();
            exApp.Visible = true;
            foreach (Office.COMAddIn item in exApp.COMAddIns)
                Console.WriteLine(item.Description);

            Excel.Workbook book = (Excel.Workbook)exApp.Workbooks.Add();
            foreach (Excel.Worksheet item in book.Worksheets)
            {
                Console.WriteLine(item.Name);
                Excel.Range range = (Excel.Range)item.Cells[2, 2];
            }

            exApp.Quit();
            exApp.Dispose();
        }

        /// <summary>
        /// Clean up any resources being used.
        /// </summary>
        /// <param name="disposing">true if managed resources should be disposed; otherwise, false.</param>
        protected override void Dispose(bool disposing)
        {
            if (disposing && (components != null))
            {
                components.Dispose();
            }
            base.Dispose(disposing);
        }

        #region Windows Form Designer generated code

        /// <summary>
        /// Required method for Designer support - do not modify
        /// the contents of this method with the code editor.
        /// </summary>
        private void InitializeComponent()
        {
            this.SuspendLayout();
            // 
            // Form1
            // 
            this.ClientSize = new System.Drawing.Size(292, 273);
            this.Name = "Form1";
            this.Text = "ClientApplication";
            this.ResumeLayout(false);

        }

        #endregion
    }
}

File: LateBindingApi.CodeGenerator.VB/TestClient/Form1.vb
Imports NetOffice
Imports xyz

Public Class Form1

    Dim _excelApp As Excel.Application
    Dim WithEvents _workBook As Excel.Workbook

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click

        Factory.Initialize()

        _excelApp = New Excel.Application()
        _excelApp.Visible = True
        _workBook = _excelApp.Workbooks.Add()

        Dim worksheet As Excel.Worksheet = _workBook.Worksheets.Add()
         
    End Sub

    Private Sub _workBook_BeforeCloseEvent(ByRef Cancel As Boolean) Handles _workBook.BeforeCloseEvent
        Cancel = True
    End Sub

    Private Sub Button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button2.Click
        _excelApp.Quit()
        _excelApp.Dispose()
    End Sub
End Class
